src/amo/components/DownloadFirefoxButton/index.js

```diff
@@ -0,0 +1,57 @@
+/* @flow */
+import classNames from 'classnames';
+import React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import translate from 'core/i18n/translate';
+import type { UserAgentInfoType } from 'core/reducers/api';
+import Button from 'ui/components/Button';
+
+
+type PropTypes = {
+  className?: 'string',
+  i18n: Object,
+  userAgentInfo: UserAgentInfoType,
+}
+
+export const DownloadFirefoxButtonBase = ({
+  className,
+  i18n,
+  userAgentInfo,
+// eslint-disable-next-line indent
+}: PropTypes = {}) => {
+  if (userAgentInfo.browser.name === 'Firefox') {
+    return null;
+  }
+
+  return (
+    <Button
+      className={classNames(
+        'DownloadFirefoxButton',
+        'Button',
+        'Button--confirm',
+        'Button--small',
+        className,
+      )}
+      href="http://mozilla.org/"
+    >
+      {i18n.gettext('Download Firefox')}
+    </Button>
+  );
+};
+
+type StateType = {|
+  api: { userAgentInfo: UserAgentInfoType },
```

Comment:
This should import [`ApiStateType`](https://github.com/mozilla/addons-frontend/blob/master/src/core/reducers/api.js#L31) and change to:

```js
type StateType = {|
  api: ApiStateType,
|}
```

You could also just do:
```js
export function mapStateToProps(state: {| api: ApiStateType |}) {
  // ...
}
```

Reply:
Ah, right on.

Reply:
PS I definitely prefer the syntax/style of keeping the type definition out of the function signature. As soon as I see the curly braces my brain thinks object destructing and I have to double-take. If we stick to using colons and the type only in the function signatures it's actually quite easy to see what is/isn't flow.

Reply:
> PS I definitely prefer the syntax/style of keeping the type definition out of the function signature.

same here